cloudify/amqp_client.py

```diff
@@ -266,34 +266,17 @@ def add_handler(self, handler):
             handler.register(self._pika_connection, self._publish_queue)
 
 
-class TaskConsumer(object):
-    routing_key = ''
+def agent_queue_name(queue, routing_key):
+    return '{0}_{1}'.format(queue, routing_key)
 
-    def __init__(self, queue, threadpool_size=5):
-        self.threadpool_size = threadpool_size
-        self.exchange = queue
-        self.queue = '{0}_{1}'.format(queue, self.routing_key)
-        self._sem = threading.Semaphore(threadpool_size)
-        self._output_queue = None
-        self.in_channel = None
 
-    def register(self, connection, output_queue):
-        self._output_queue = output_queue
-        self.in_channel = connection.channel()
-        self._register_queue(self.in_channel)
+class ProcessMixin(object):
+    """Implement processing incoming messages using a threadpool"""
 
-    def _register_queue(self, channel):
-        channel.basic_qos(prefetch_count=self.threadpool_size)
-        channel.confirm_delivery()
-        channel.exchange_declare(
-            exchange=self.exchange, auto_delete=False, durable=True)
-        channel.queue_declare(queue=self.queue,
-                              durable=True,
-                              auto_delete=False)
-        channel.queue_bind(queue=self.queue,
-                           exchange=self.exchange,
-                           routing_key=self.routing_key)
-        channel.basic_consume(self.process, self.queue)
+    def __init__(self, threadpool_size=5, *args, **kwargs):
+        self.threadpool_size = threadpool_size
+        self._sem = threading.Semaphore(self.threadpool_size)
+        super(ProcessMixin, self).__init__(*args, **kwargs)
```

Comment:
Let's add a comment here re: this super call, so that posterity won't try to remove it as unnecessary.